src/ethereum/ethEngine.js

```diff
@@ -709,6 +711,28 @@ export class EthereumEngine extends CurrencyEngine {
     }
     return ''
   }
+
+  // Overload saveTx to mutate replaced transactions by RBF
+  async saveTx(edgeTransaction: EdgeTransaction) {
+    // We must check if this transaction replaces another transaction
+    if (edgeTransaction.otherParams && edgeTransaction.otherParams.rbfTxid) {
+      const { currencyCode } = this.currencyInfo
+
+      // Get the replaced transaction using the rbfTxid
+      const txid = edgeTransaction.otherParams.rbfTxid
+      const idx = this.findTransaction(currencyCode, txid)
+      const replacedEdgeTransaction = this.transactionList[currencyCode][idx]
+
+      // Update the transaction's blockHeight to -1 (drops the transaction)
+      const updatedEdgeTransaction: EdgeTransaction = {
+        ...replacedEdgeTransaction,
+        blockHeight: -1
+      }
+      this.updateTransaction(currencyCode, updatedEdgeTransaction, idx)
```

Comment:
This should actually use addTransaction. Yes, it's a poor name but add transaction will also update transactions if the transaction already exists. It also does additional validation necessary

Reply:
Amended commit and tested it using addTransaction.